tools/project.py

```diff
@@ -898,6 +946,34 @@ def write(self, n: ninja_syntax.Writer) -> None:
         source_inputs: List[Path] = []
         source_added: Set[Path] = set()
 
+        all_pch_outputs = []
+        if config.precompiled_headers:
+            for pch in config.precompiled_headers:
+                pch_out_name = Path(pch["source"]).with_suffix(".mch")
+
+                n.comment(f"Precompiled header {pch_out_name}")
+                pch_out_abs_path = Path(get_pch_out_name(config, pch))
+                n.build(
+                    outputs=pch_out_abs_path,
+                    rule="mwcc_pch",
```

Comment:
We should conditionally use the `_sjis` rule here, using some logic like:
`pch.get("shift_jis", config.shift_jis)` so that we fall back to the global config if `shift_jis` wasn't specified